### PR TITLE
Updated XRM.Navigation.navigateTo to be up to date

### LINKED ISSUE #182: Typings for XRM.Navigation.navigateTo are out of date.
**Is your feature request related to a problem? Please describe.**
Xrm.Navigation.navigateTo does not provide full typings for `pageInput.pageType`. It only supports `"entitylist"` and `"webresource"`, but not `"entityrecord"`.

The fields used only for `pageType: "entityrecord"` are also missing typings. These include: `entityId`, `createFromEntity`, `formId`, `isCrossEntityNavigation`, `isOfflineSyncError`, `processId`, `processInstanceId`, `relationship`, and `selectedStageId`.

**Describe the solution you'd like**
Updated typings.

**Describe alternatives you've considered**
Using Xrm.Navigation.openForm instead.
Writing untyped code (just kidding).

**Additional context**
[Microsoft's documentation](https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/xrm-navigation/navigateto)

### PR DESCRIPTION
**Pull request by KnethM**
Fixes #182

### CODE REVIEW

**Comment by sudhip-git** [comment]
Could you please confirm when this pull request will be merged

**Comment by magesoe** [comment]
I am back from vacation next week, will look at this PR then

File: src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts [hunk 1]
```diff
@@ -499,7 +560,7 @@ declare namespace Xrm {
          * @param pageInput Input about the page to navigate to. The object definition changes depending on the type of page to navigate to: entity list or HTML web resource.
          * @param navigationOptions Options for navigating to a page: whether to open inline or in a dialog. If you don't specify this parameter, page is opened inline by default.
          */
-        navigateTo(pageInput: PageInput, navigationOptions: NavigationOptions): Promise<undefined>;
+        navigateTo(pageInput: EntityRecord | EntityList | WebResource, navigationOptions: NavigationOptions): Promise<undefined>;
```
Comment (magesoe) [inline src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts]: According to the documentation, navigationOptions is optional, so it should be navigationOptions?

File: src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts [hunk 2]
```diff
@@ -411,22 +411,19 @@ declare namespace Xrm {
         savedEntityReference: Lookup[];
     }
 
-    type PageType = "entitylist" | "webresource";
+    type PageType = "entitylist" | "entityrecord" | "webresource";
```
Comment (magesoe) [inline src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts]: Love the idea you had of extending a common interface, however it does pose a problem. Right now there is no connection between the chosen PageType and the type of extension. This means you are allowed to pass false parameters as this
```typescript
Xrm.Navigation.navigateTo({
      pageType: "entitylist",
      entityName: "",
      createFromEntity: {
        entityType: "",
        id: "",
        name: ""
      }
    })
```
The `pageType` variable should be moved to each declaration with a single allowed string that matches that declaration. For example `interface EntityList` should look like this 
```typescript
interface EntityList {
    pageType: "entitylist"
    ...
}
```
The reason why the base interface is removed is that pageType was the only common type among these interfaces. You should still keep the navigateTo function the same, with the union of the three interfaces.

**Review by magesoe — CHANGES_REQUESTED** [review]
Love the PR, some minor changes. Do you agree with them?

**Review by magesoe — APPROVED** [review]

**Pull request merged**